common_examples/tcp_client/main/tcp_client.c

```diff
@@ -33,96 +30,124 @@ static void got_ip_event_handler(void *arg, esp_event_base_t event_base, int32_t
     ESP_LOGI(TAG, "ETHMASK:" IPSTR, IP2STR(&ip_info->netmask));
     ESP_LOGI(TAG, "ETHGW:" IPSTR, IP2STR(&ip_info->gw));
     ESP_LOGI(TAG, "~~~~~~~~~~~");
-    xSemaphoreGive(x_got_ip_semaphore);
+    xSemaphoreGive(got_ip_sem);
 }
 
 void app_main(void)
 {
     // Create default event loop that running in background
     ESP_ERROR_CHECK(esp_event_loop_create_default());
     // Initialize semaphore
-    x_got_ip_semaphore = xSemaphoreCreateBinary();
+    got_ip_sem = xSemaphoreCreateBinary();
+    if (got_ip_sem == NULL) {
+        ESP_LOGE(TAG, "Failed to create semaphore");
+        return;
+    }
+
     // Initialize Ethernet driver
     uint8_t eth_port_cnt = 0;
     esp_eth_handle_t *eth_handles;
-    char if_key_str[10];
-    char if_desc_str[10];
     ESP_ERROR_CHECK(ethernet_init_all(&eth_handles, &eth_port_cnt));
```

Comment:
Shouldn't we add a warning the we'll use only the first instance? (unless i missed it, you just use `eth_hanldes[0]` below in the code)?

Reply:
Good point! I messed up! The goal is to use multiple interfaces.